File: Chavah.NetCore/wwwroot/js/Models/Album.ts
namespace BitShuva.Chavah {
    export class Album implements Server.Album {
        artist: string;
        name: string;
        albumArtUri: string | null;
        id: string;
        backgroundColor: string;
        foregroundColor: string;
        mutedColor: string;
        textShadowColor: string;
        isVariousArtists: boolean;
        songCount: number;

        isSaving = false;

        constructor(serverObj: Server.Album) {
            angular.merge(this, serverObj);
        }
    }
}

File: Chavah.NetCore/Models/Album.cs
﻿using System;
using BitShuva.Chavah.Common;

namespace BitShuva.Chavah.Models
{
    /// <summary>
    /// An album for which album art has been uploaded.
    /// </summary>
    public class Album
    {
        public string Artist { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? HebrewName { get; set; }
        public Uri AlbumArtUri { get; set; } = UriExtensions.Localhost;
        public string? Id { get; set; }
        public string BackgroundColor { get; set; } = string.Empty;
        public string ForegroundColor { get; set; } = string.Empty;
        public string MutedColor { get; set; } = string.Empty;
        public string TextShadowColor { get; set; } = string.Empty;
        public bool IsVariousArtists { get; set; }
        public int SongCount { get; set; }
    }
}
